tests/Backend/SOX/BranchStorageTest.php

```diff
@@ -0,0 +1,264 @@
+                $branches[] = $branch;
+            }
+        }
+        return $branches;
+    }
+
+    public function testDeleteTable(): void
+    {
+        [$privilegedClient, $productionTableId, $branchClient, $devTableId] = $this->initResources();
+
+        // drop branch table
+        $branchClient->dropTable($devTableId);
+
+        // check that production table still exists and we can preview data
+        $privilegedClient->getTable($productionTableId);
+        $privilegedClient->getTableDataPreview($productionTableId);
+
+        try {
+            $branchClient->getTable($devTableId);
+            $this->fail('Table should not exist');
+        } catch (ClientException $e) {
+            $this->assertSame(404, $e->getCode());
+        }
+
+        // pull table again
+        $devTableId = $branchClient->pullTableToBranch($productionTableId);
+
+        // drop production table
+        $privilegedClient->dropTable($devTableId);
+
+        // check that development table still exists and we can preview data
+        $branchClient->getTable($productionTableId);
+        $branchClient->getTableDataPreview($productionTableId);
+
+        try {
+            $privilegedClient->getTable($devTableId);
+            $this->fail('Table should not exist');
+        } catch (ClientException $e) {
+            $this->assertSame(404, $e->getCode());
+        }
+    }
+
+    public function testTableImportExport(): void
+    {
+        [$privilegedClient, $productionTableId, $branchClient, $devTableId] = $this->initResources();
+        $this->assertTableRowsCount(5, $devTableId, $branchClient);
+        $this->assertTableRowsCount(5, $productionTableId, $privilegedClient);
+
+        $branchClient->writeTableAsync(
+            $devTableId,
+            new CsvFile(__DIR__ . '/../../_data/languages.increment.csv'),
+            [
+                'incremental' => true,
+            ]
+        );
+        $this->assertTableRowsCount(8, $devTableId, $branchClient);
+        $this->assertTableRowsCount(5, $productionTableId, $privilegedClient);
+
+        $privilegedClient->writeTableAsync(
+            $productionTableId,
+            new CsvFile(__DIR__ . '/../../_data/languages.more-rows.csv'),
+            [
+                'incremental' => true,
+            ]
+        );
+        $this->assertTableRowsCount(8, $devTableId, $branchClient);
+        $this->assertTableRowsCount(12, $productionTableId, $privilegedClient);
+
+        // test export
+        $results = $branchClient->exportTableAsync($devTableId);
+        // @todo SOX-20 will save file to branch storage and we can use $branchClient but for now we need to use $privilegedClient
+        $this->assertFileRowsCount(8, $results['file']['id'], $privilegedClient);
+
+        $results = $privilegedClient->exportTableAsync($productionTableId);
+        $this->assertFileRowsCount(12, $results['file']['id'], $privilegedClient);
+    }
+
+    public function testTableOperations(): void
```

Comment:
nechtel bys to prosim rozdelit podle usecasu? jakoze ano, muze to byt jeden velkej test, ale pak nam pada visibilita.

Reply:
rozděleno